echopype/tests/test_preprocess.py

```diff
@@ -0,0 +1,151 @@
+    null = ds_Sv.Sv_corrected.isnull()
+    # Test to see if the right number of points are removed before the range gets too large
+    assert np.count_nonzero(null.isel(frequency=0, range_bin=slice(None, 50))) == 6
+
+
+def test_compute_MVBS_index_binning():
+    """Test compute_MVBS_index_binning on toy data"""
+
+    # Parameters for fake data
+    nfreq, npings, nrange = 4, 40, 400
+    ping_num = 2             # number of pings to average over
+    range_bin_num = 3        # number of range_bins to average over
+
+    # Construct data with values that increase every ping_num and range_bin_num
+    # so that when compute_MVBS_index_binning is performed, the result is a smaller array
+    # that increases by 1 for each row and column
+    data = np.zeros((nfreq, npings, nrange))
+    for p_i, ping in enumerate(range(0, npings, ping_num)):
+        for r_i, rb in enumerate(range(0, nrange, range_bin_num)):
+            data[0, ping:ping + ping_num, rb:rb + range_bin_num] += r_i + p_i
+    for f in range(nfreq):
+        data[f] = data[0]
+
+    data_log = 10 * np.log10(data)      # Convert to log domain
+    freq_index = np.arange(nfreq)
+    ping_index = np.arange(npings)
+    range_bin = np.arange(nrange)
+    Sv = xr.DataArray(data_log, coords=[('frequency', freq_index),
+                                        ('ping_time', ping_index),
+                                        ('range_bin', range_bin)])
+    Sv.name = "Sv"
+    ds_Sv = Sv.to_dataset()
+    ds_Sv = ds_Sv.assign(range=xr.DataArray(np.array([[np.linspace(0, 10, nrange)] * npings] * nfreq), coords=Sv.coords))
+
+    # Binned MVBS test
+    ds_MVBS = ep.preprocess.compute_MVBS_index_binning(
+        ds_Sv,
+        range_bin_interval=range_bin_num,
+        ping_num_interval=ping_num
+    )
+    # Shape test
+    shape = (nfreq, npings / ping_num, nrange / range_bin_num)
+    assert np.all(ds_MVBS.Sv.shape == np.ceil(shape))
+
+    data_test = (10 ** (ds_MVBS.Sv / 10)).round().astype(int)    # Convert to linear domain
+    # Test values along range_bin
+    assert np.all(data_test.isel(frequency=0, ping_time=0) == np.arange(nrange / range_bin_num))
+    # Test values along ping time
+    assert np.all(data_test.isel(frequency=0, range_bin=0) == np.arange(npings / ping_num))
+
+
+def test_compute_MVBS():
+    """Test compute_MVBS on toy data"""
+
+    # Parameters for fake data
+    nfreq, npings, nrange = 1, 60, 600
+    range_meter_bin = 5          # range in meters to average over
+    ping_time_bin = 5            # number of seconds to average over
+    ping_rate = 2                # Number of pings per second
+    range_bin_per_meter = 30     # Number of range_bins per meter
+    ping_num = npings // ping_rate // ping_time_bin                      # number of pings to average over
+    range_bin_num = nrange // range_bin_per_meter // range_meter_bin     # number of range_bins to average over
+    total_range = nrange // range_bin_per_meter
```

Comment:
To test the `test_compute_MVBS` properly I think it'll be useful to make some changes to the toy data.
- use `nfreq` > 1
- make the total number of pings and the total number of range_bins **not** multiples of the number of pings and range_bins to average over, so that we know if the `groupby_bins(...).resample(...).mean()` operation is done correctly.